packages/frontend/core/src/modules/docs-search/entities/docs-indexer.ts

```diff
@@ -0,0 +1,299 @@
+import { DebugLogger } from '@affine/debug';
+import type { Job, WorkspaceService } from '@toeverything/infra';
+import {
+  Document,
+  Entity,
+  IndexedDBIndexStorage,
+  IndexedDBJobQueue,
+  JobRunner,
+} from '@toeverything/infra';
+import { difference } from 'lodash-es';
+import type { Array as YArray, Map as YMap } from 'yjs';
+import { applyUpdate, Doc as YDoc } from 'yjs';
+
+import { blockIndexSchema, docIndexSchema } from '../schema';
+
+const logger = new DebugLogger('crawler');
+
+interface IndexerJobPayload {
+  docId: string;
+}
+
+export class DocsIndexer extends Entity {
+  private readonly jobQueue = new IndexedDBJobQueue<IndexerJobPayload>(
```

Comment:
this is exactly what I mean for higher abstraction for indexer, without refer to specific backend implementation

Reply:
if we have sqlite based indexer for electron, this would be hard to adapt

Reply:
then depends on interface instead of implementation

Reply:
I'll do this when we have other implementations, but for now I've added a type here.